update.go

```diff
@@ -319,8 +319,9 @@ other options are ignored.
 		config.Cgroups.Resources.MemoryReservation = *r.Memory.Reservation
 		config.Cgroups.Resources.MemorySwap = *r.Memory.Swap
 		config.Cgroups.Resources.MemoryCheckBeforeUpdate = *r.Memory.CheckBeforeUpdate
-		config.Cgroups.Resources.PidsLimit = r.Pids.Limit
 		config.Cgroups.Resources.Unified = r.Unified
+		l := r.Pids.Limit
+		config.Cgroups.Resources.PidsLimit = &l
```

Comment:
I think maybe we can use `config.Cgroups.Resources.PidsLimit = &r.Pids.Limit` or `*config.Cgroups.Resources.PidsLimit = r.Pids.Limit`?